components/audioinput/microphone/microphone.go

```diff
@@ -77,30 +77,38 @@ func newMicrophoneSource(attrs *Attrs, logger golog.Logger) (audioinput.AudioInp
 			return nil, err
 		}
 	}
-
-	labels := gostream.QueryAudioDeviceLabels()
-	for _, label := range labels {
-		if debug {
-			logger.Debugf("%s", label)
-		}
-
-		if pattern != nil && !pattern.MatchString(label) {
-			if debug {
-				logger.Debug("\t skipping because of pattern")
+	all := gostream.QueryAudioDevices()
+
+	for i, info := range all {
+		logger.Debugf("%s", info.ID)
+		logger.Debugf("\t labels: %v", info.Labels)
+		for _, p := range info.Properties {
+			logger.Debugf("\t %+v", p.Audio)
+			if pattern != nil && !pattern.MatchString(info.Labels[i]) {
+				if debug {
+					logger.Debug("\t skipping because of pattern")
+				}
+				continue
 			}
-			continue
-		}
-
-		s, err := tryMicrophoneOpen(label, gostream.DefaultConstraints, logger)
-		if err == nil {
-			if debug {
-				logger.Debug("\t USING")
+			if p.Audio.ChannelCount > 0 {
```

Comment:
What exactly is `ChannelCount`?

Reply:
For sound files, it represents stereo vs mono. In other words, if  "left" and "right" in headphones are the same. If there is only one channel, it means "left" is the same as "right". The driver that was selected before had none, so it was a way to find a not empty driver

Reply:
Ah, I see. So this check will ensure we're using an audio driver that emits sound rather than one that doesn't like before?

Reply:
Yes exactly